src/components/SpotifyPlayer.vue

```diff
@@ -80,25 +80,18 @@
             playTrack(track) {
                 const deviceId = this.player._options.id
 
-                axios.put(`https://api.spotify.com/v1/me/player/play?device_id=${deviceId}`,
-                    JSON.stringify({uris: [track]}),
-                    {
-                        headers: {
-                            'Content-Type': 'application/json',
-                            'Authorization': `Bearer ${this.token.token}`
-                        },
-                    })
+                SpotifyService.playTrack(deviceId, track, this.token.token)
+                    .catch(console.error) // eslint-disable-line no-console
             },
             createPlayer() {
                 window.console.log('creating player...')
 
-                // eslint-disable-next-line
-                this.player = new Spotify.Player({
-                    name: 'My Spotify Player',
-                    getOAuthToken: cb => { cb(this.token.token); }
-                });
+                this.player = SpotifyService.createPlayer(
+                    'My Spotify Player',
+                    this.token.token,
+                )
 
-                this.player.addListener('ready', ({device_id}) => {
+                this.player.on('ready', ({device_id}) => {
```

Comment:
Looks like `addListener` and `on` do the same thing, so I switch this to make my test mocking easier, and to make this block more consistent.